File: solutions/julia/cars-assemble/1/cars-assemble.jl
function success_rate(speed)
    if speed == 0
        0
    elseif 1 <= speed <= 4
        1
    elseif 5 <= speed <= 8
        0.9
    elseif speed == 9
        0.8
    else
        0.77
    end
end

function production_rate_per_hour(speed)
    speed * 221 * success_rate(speed)
end

function working_items_per_minute(speed)
    Int(div(production_rate_per_hour(speed), 60))
end

File: solutions/csharp/cars-assemble/1/CarsAssemble.cs
static class AssemblyLine
{
    public static double SuccessRate(int speed)
    {
        if (speed == 0)
        {
            return 0.0;
        } else if (1 <= speed && speed <= 4)
        {
            return 1.0;
        } else if (5 <= speed && speed <= 8)
        {
            return 0.9;
        } else if (speed == 9)
        {
            return 0.8;
        } else
        {
            return 0.77;
        }
    }
    
    public static double ProductionRatePerHour(int speed)
    {
        return speed * 221.0 * AssemblyLine.SuccessRate(speed);
    }

    public static int WorkingItemsPerMinute(int speed)
    {
        return (int)(AssemblyLine.ProductionRatePerHour(speed)) / 60;
    }
}
